raft/broadcast_tables/experimental/lang.hh

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+#pragma once
+
+#include <optional>
+#include <variant>
+
+#include "bytes.hh"
+
+
+namespace raft::broadcast_tables {
+
+using namespace std::string_literals;
+
+// Represents 'SELECT value WHERE key = {key} FROM system.broadcast_kv_store;'.
+struct select_query {
+    bytes key;
+};
+
+// Represents 'UPDATE system.broadcast_kv_store SET value = {new_value} WHERE key = {key} [IF value = {value_condition}];'.
+// If value_condition is nullopt, the update is unconditional.
+struct update_query {
+    bytes key;
+    bytes new_value;
+    std::optional<bytes_opt> value_condition;
```

Comment:
The condition is that the value matches some old value, so a more accurate name is old_value or some such. I'd expect a variable named "condition" to contain an expression, not a value. Also bytes_opt -> managed_bytes_opt.

The evaluation of a NULL here should match nothing. So if the outer optional is engaged and the inner optional is disengaged. The query is not applied.

Reply:
> The evaluation of a NULL here should match nothing. So if the outer optional is engaged and the inner optional is disengaged. The query is not applied.

But that's not how LWT works, IIRC. And we don't want to introduce different semantics.
In LWT when you do `IF v = null` the update will be applied iff the column was `null`.